tests/test_registry.py

```diff
@@ -645,6 +645,58 @@ def raise_exception():
         assert "error" in response
         assert random_val == response["error"]["data"]["random"]
 
+    def test_encoder_exception_batch(self):
+        def custom_encode(input):
+            if "foo" in str(input):
+                raise InternalError("Could not parse the input data.")
+            else:
+                return json.JSONEncoder().encode(input)
+
+        registry = Registry()
+        registry.json_encoder.encode = custom_encode
+
+        fake_batch_request = self._create_fake_request([{
+            "jsonrpc": "2.0",
+            "method": "rpc.describe",
+            "params": [],
+            "id": "foo",
+        }, {
+            "jsonrpc": "2.0",
+            "method": "rpc.describe",
+            "params": [],
+            "id": "bar",
+        }])
+        response = json.loads(registry.dispatch(fake_batch_request))
+
+        assert isinstance(response, list)
+        assert len(response) == 2
+
+        foo_response = response[0] if response[1]["id"] == "bar" else response[1]
+        TestDispatch.assert_error(foo_response, "foo", InternalError)
+        assert foo_response["error"]["data"] == "Could not parse the input data."
+
+        bar_response = response[1] if response[1]["id"] == "bar" else response[0]
```

Comment:
Can you put the response checking together:

    if response[0]["id"] == "foo":
        foo_response = response[0]
        bar_response = response[1]
   else:
        ...